File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int right = people.length - 1;
        int left = 0;
        int result = 0;
        
        while (left <= right){
            if(people[left] + people[right] <= limit){
                left++;
            }
            result++;
            right--;
        }
        
        return result;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a,b) => a - b);
    let cnt = 0;
    let first = 0;
    let last = people.length - 1
    while (first < last){
        if (people[first] +people[ last] <= limit){
            cnt += 1;
            first += 1; last -= 1;
            continue;
        }
        
        last -= 1;
        cnt += 1;
    }
    
    return cnt + (first === last ? 1 : 0)
    
}